web/modules/custom/ai_screening_reports/src/Controller/AiScreeningReportsController.php

```diff
@@ -66,26 +68,30 @@ public function __invoke(NodeInterface $node, Request $request): array {
   public function projectTrack(Request $request): array|RedirectResponse {
     $loopCounter = 0;
     $projectTracks = $this->projectTrackHelper->loadTracks((array) $request->get('project_track_id'));
+
     // Ensure proper url parameters: ?project_track_id[]=1&project_track_id[]=3.
     if (!empty($projectTracks)) {
+      // Allow the first term to define the dimensions and the thresholds.
+      $term = reset($projectTracks)->getType();
+
+      /** @var \Drupal\taxonomy\TermInterface $term */
+      $dimensions = $this->projectTrackTypeHelper->getDimensions($term);
       $projectData = [
-        // @todo get thresholds.
         'thresholds' => [
-          'x' => 25,
-          'y' => 25,
-          'z' => 25,
+          'x' => $this->projectTrackTypeHelper->getThreshold($term->id(), 0, 'approved') ?? '',
```

Comment:
You seem to be mixing number and string values here. Consider using `null` (or `0`) for “no value”.

Reply:
getThreshold now only accepts int for the first two params and string(Enum) for the third and returns int